### DISCUSSION TITLE
awscli (on Linux, at least) seems to want to install some radical deps

### DISCUSSION THREAD

**Discussion by rjray** [Homebrew/discussions#169] [Everyday usage]
The last time I updated `awscli`, it ended up installing a lot of extra stuff to satisfy the `groff` dependency. Today I saw another update, so I started an upgrade and now it wants to install `freeglut` as a dependency. Is there something about a CLI tool needing FreeGLUT that I don't quite understand?

**Comment by iMichka** [comment]
You can have a look at the dependency tree like this:
`brew deps --tree awscli`:
awscli -> groff -> netpbm -> jasper -> freeglut

Of course, if you think that groff is not needed for awscli, and we can build without it, a pull request is welcome. We often add dependencies because the build fails without it. But we are open to reduce the size of the dependency tree if we can.

**Comment by rjray** [comment]
Unfortunately, I don't have the background to diagnose a homebrew recipe. I just find it odd that the package needs `groff` to run. To build, perhaps, but to run? I generally install from bottles, so I should have only needed run-time dependencies, right?

**Reply by carlocab** [reply]
The docs suggest that `groff` is a runtime dependency: https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html

Admittedly, the docs don't explicitly say that it's a runtime dependency, only that it is a prerequisite. However, their installation instructions just involve an install script that doesn't build anything, so I doubt they would list it as a prerequisite if it were only a build dependency.

**Reply by gromgit** [reply]
> I just find it odd that the package needs `groff` to run.

The docs say that text table output is an option. Odds are good that `awscli` "outsources" this to `tbl`, which is part of `groff`.
